dsp/modules/hf_client.py

```diff
@@ -128,47 +128,89 @@ def __init__(self, model, port, url="http://localhost", **kwargs):
         
         self.headers = {"Content-Type": "application/json"}
         self.kwargs |= kwargs
+        # kwargs needs to have model, port and url for the lm.copy() to work properly
+        self.kwargs.update({
+            'model': model,
+            'port': port,
+            'url': url
+        })
 
 
     def _generate(self, prompt, **kwargs):
         kwargs = {**self.kwargs, **kwargs}
-
-        payload = {
-            "model": self.kwargs["model"],
-            "prompt": prompt,
-            **kwargs,
-        }
-
         
+        # get model_type
+        model_type = kwargs.get("model_type",None)
```

Comment:
let's actually move this to the initialization layer so the user can specify the model_type when configuring the vLLM client, not during generation time.